File: CivitasRuby/Casilla_Calle.rb
module Civitas

    class Casilla_Calle < Casilla
        attr_reader :titulo

        def initialize(titulo)
            super(titulo.nombre)
            @titulo=titulo
            @importe=titulo.precioCompra
        end

        def recibeJugador(actual,todos)
            if jugadorCorrecto(actual, todos)
            
                informe(actual, todos)
                jugador=todos[actual]

                if !@titulo.tienePropietario
                    
                    jugador.puedeComprarCasilla
                else
                    titulo.tramitarAlquiler(jugador)
                end
            end
        end

        def toString
            
            ret=@nombre + ' de tipo calle '
            ret+=@titulo.toString
            return ret
            
        end

    end
    
end

File: CivitasJava/civitas/Casilla_Calle.java
package civitas;

import java.util.ArrayList;

class Casilla_Calle extends Casilla
{
    private TituloPropiedad titulo;
    private float importe;

    Casilla_Calle(TituloPropiedad titulo)
    {
        super(titulo.getNombre());
        this.titulo=titulo;
        this.importe=titulo.getPrecioCompra();
    }

    TituloPropiedad getTituloPropiedad() 
    {
        return this.titulo;
    }
    
    @Override
    void recibeJugador(int iactual, ArrayList<Jugador> todos)
    {   
        if (super.jugadorCorrecto(iactual, todos)) 
        {
            super.informe(iactual, todos);
            Jugador jugador = todos.get(iactual);
            if (!this.titulo.tienePropietario()) {
                jugador.puedeComprarCasilla();
            } else {
                this.titulo.tramitarAlquiler(jugador);
            }
        }
    }

    @Override
    public String toString() 
    {
        String ret = getNombre() + " de tipo calle \n" + this.titulo.toString();
        return ret;
    }
}